tests/testthat/test-col_plan.R

```diff
@@ -0,0 +1,281 @@
+
+test_that("Defining the spanning structure", {
+
+  s1 <- span_structure(
+    label = "Test Label",
+    vars(A,B)
+  )
+
+  s2 <- span_structure(
+    label = "Test Label",
+    span_structure(
+      label = "Test Sub Label",
+      vars(A,B)
+    ),
+    vars(C,D)
+  )
+
+  expect_s3_class(s1,"span_structure")
+  expect_s3_class(s2,"span_structure")
+  expect_s3_class(s2,"span_structures")
+
+  expect_equal(s1$label, "Test Label")
+  expect_equal(s2$label, "Test Label")
+  expect_equal(s2$label, "Test Label")
+
+  expect_equal(s1$span_cols, list(vars(A,B)), ignore_attr = TRUE)
+  expect_equal(s2$span_cols, list(span_structure(label = "Test Sub Label",vars(A,B)),vars(C,D)), ignore_attr = TRUE)
+
+})
+
+test_that("Defining the col plan", {
+
+  s1 <- col_plan(
+    span_structure(
+      label = "Test Label",
+      vars(A,B)
+    ))
+
+  expect_s3_class(s1,"col_plan")
+  expect_s3_class(s1,"plan")
+
+  #Go into object
+  tfrmt_test <- tfrmt(column = vars(columns),
+                      col_plan = col_plan(span_structure("test label", vars(col3))))
+
+  expect_true(!is.null(tfrmt_test$col_plan))
+
+  # expect error when writing invalid col_plan
+    expect_error(
+      tfrmt(
+        col_plan = col_plan(
+      group,label,
+      col1, col2,
+      span_structure("test label", col3),
+      col4:col10
+    )))
+
+})
+
+test_that("From the spanning structure get cols to span across", {
+
+  s1 <- span_structure(
+    label = "Test Label",
+    vars(A,B)
+  )
+
+  s1 <- span_structure(
+    label = "Test Label",
+    vars(A,B)
+  )
+
+  s2 <- span_structure(
+    label = "Test Label",
+    span_structure(
+      label = "Test Sub Label",
+      vars(A,B)
+    ),
+    vars(C,D)
+  )
+
+  s3 <- span_structure(
+    label = "Test Label",
+    vars(D),
+    span_structure(
+      label = "Test Sub Label",
+      vars(C,A)
+    ),
+    vars(B)
+  )
+
+  sample_df <- data.frame(A = character(), B = character(), C = character(), D = character())
+
+  s1_cols <- span_col_select(s1, data = sample_df)
```

Comment:
can we add a test representing column removal as well?

Reply:
I have added this, but we probably need to update these tests anyway cause the